# Initial  review of HESA Student Data

In [1]:
import pandas as pd

In [4]:
student_df = pd.read_excel("HESA Data - Personal Characteristics.xlsx", skiprows=18)
student_df.head()

# skiprows is used to remove the data information and guidance above the table in the Excel spreadsheet.

Category Marker Category First year marker Level of study Mode of study  \
0             Sex   Female               All            All           All   
1             Sex   Female               All            All           All   
2             Sex   Female               All            All           All   
3             Sex   Female               All            All           All   
4             Sex   Female               All            All           All   

  Country of HE provider            Domicile Academic Year     Number  \
0                    All                 All       2016/17  1344635.0   
1                    All             England       2016/17   906090.0   
2                    All      European Union       2016/17    76070.0   
3                    All  Non-European Union       2016/17   165460.0   
4                    All    Northern Ireland       2016/17    35870.0   

   Percentage  
0        0.57  
1        0.57  
2        0.55  
3        0.53  
4        0.57

In [5]:
student_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 386780 entries, 0 to 386779
Data columns (total 10 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Category Marker         386780 non-null  object 
 1   Category                386780 non-null  object 
 2   First year marker       386780 non-null  object 
 3   Level of study          386780 non-null  object 
 4   Mode of study           386780 non-null  object 
 5   Country of HE provider  386780 non-null  object 
 6   Domicile                386780 non-null  object 
 7   Academic Year           386780 non-null  object 
 8   Number                  328486 non-null  float64
 9   Percentage              275002 non-null  float64
dtypes: float64(2), object(8)
memory usage: 29.5+ MB


- There are 10 columns and 386780 rows in the dataset. First year marker, level of study, mode of study, country of HE provider  and Domicile all have a value of 'All' so different values are being grouped together.
- The data is measured by an academic year in the academic year column.
- There is a number and percentage column which refers to the student numbers and corresponding percentages for the row.
- There are two different data types in the dataset - float and object and there are no null values in any of the columns.


In [6]:
student_df.describe()

Number     Percentage
count  3.284860e+05  275002.000000
mean   1.704747e+04       0.325762
std    8.918171e+04       0.359019
min    0.000000e+00       0.000000
25%    1.000000e+01       0.020000
50%    1.750000e+02       0.150000
75%    2.825000e+03       0.570000
max    2.751865e+06       1.000000

- Statistical summary data for student number and percentage

In [9]:
student_df["Category Marker"].describe()

count               386780
unique                   7
top       Religious belief
freq                145735
Name: Category Marker, dtype: object

The Category Marker column contains 7 unique values and the data type is object

In [10]:
student_df["Category"].describe()

count        386780
unique           26
top       Not known
freq          30415
Name: Category, dtype: object

The Category Marker column contains 26 unique values and the data type is object. The top value shows that this column has values designated as not known.

In [12]:
student_df["First year marker"].describe()

count     386780
unique         3
top          All
freq      131595
Name: First year marker, dtype: object

The First year marker column contains 3 unique values and has data type object. The All value indicated in top shows that values in this column are being grouped together.

In [13]:
student_df["Level of study"].describe()

count     386780
unique         7
top          All
freq       59130
Name: Level of study, dtype: object

The Level of study column contains 7 unique values and has data type object. The All value indicated in top shows that values in this column are being grouped together.

In [14]:
student_df["Mode of study"].describe()

count     386780
unique         3
top          All
freq      133505
Name: Mode of study, dtype: object

The Mode of study column contains 3 unique values and has data type object. The All value indicated in top shows that values in this column are being grouped together.

In [15]:
student_df["Country of HE provider"].describe()

count     386780
unique         5
top          All
freq       86310
Name: Country of HE provider, dtype: object

The Country of HE provider column contains 5 unique values and has data type object. The All value indicated in top shows that values in this column are being grouped together.

In [16]:
student_df["Domicile"].describe()

count     386780
unique        11
top          All
freq       44000
Name: Domicile, dtype: object

The Domicile column contains 11 unique values and has data type object. The All value indicated in top shows that values in this column are being grouped together.

In [17]:
student_df["Academic Year"].describe()

count      386780
unique          5
top       2016/17
freq        77356
Name: Academic Year, dtype: object

The Academic Year column contains 5 unique values and has data type object, indicating there are 5 years worth of data in the dataset.

In [19]:
student_df.sort_values(by=['Academic Year'],ascending=True, inplace=True )
student_df

Category Marker          Category First year marker  \
0                     Sex            Female               All   
51573    Religious belief         Spiritual               All   
51572    Religious belief         Spiritual               All   
51571    Religious belief         Spiritual               All   
51570    Religious belief         Spiritual               All   
...                   ...               ...               ...   
335206  Disability Status  Known disability       Other years   
335205  Disability Status  Known disability       Other years   
335204  Disability Status  Known disability       Other years   
335202  Disability Status  Known disability       Other years   
386779                Sex         Not known       Other years   

             Level of study Mode of study Country of HE provider  \
0                       All           All                    All   
51573                   All           All                  Wales   
51572                   All           All               Scotland   
51571                   All           All               Scotland   
51570                   All           All               Scotland   
...                     ...           ...                    ...   
335206  Other undergraduate     Part-time               Scotland   
335205  Other undergraduate     Part-time               Scotland   
335204  Other undergraduate     Part-time       Northern Ireland   
335202  Other undergraduate     Part-time       Northern Ireland   
386779    All undergraduate     Part-time                England   

            Domicile Academic Year     Number  Percentage  
0                All       2016/17  1344635.0        0.57  
51573            All       2016/17        NaN         NaN  
51572          Wales       2016/17        NaN         NaN  
51571       Total UK       2016/17        NaN         NaN  
51570   Total Non-UK       2016/17        NaN         NaN  
...              ...           ...        ...         ...  
335206       England       2020/21       15.0        0.09  
335205           All       2020/21      725.0        0.21  
335204      Total UK       2020/21       80.0        0.18  
335202      Scotland       2020/21        0.0         NaN  
386779      Total UK       2020/21        0.0         NaN  

[386780 rows x 10 columns]

The data has now been sorted by Academic Year in ascending order, from 2016/7 to 2020/1.

The data now needs to be reviewed to identify information which can be removed from the dataset. We want to retain data that is relevant in answering our project questions, but remove the data which are outside the scope of the project.

**Category Marker**

Our project is focusing on gender, but there are a number of other markers which are included. For example, in the above dataframe, we can see religious belief and disability status. We need to review the data in this column.

In [20]:
# group by example
#movies_mean = movies_df.groupby(['Year','IMDB Score'], as_index=False)['Gross Earnings', 'IMDB Score'].mean()
#movies_mean
#movies_by_country_count = movies_df.groupby(['Country','Year'])['Budget'].count()
#movies_by_country_count 
#movies_mean = movies_df.groupby('Year')['Gross Earnings'].mean()
#movies_mean

In [21]:
student_df["Category Marker"].unique()

array(['Sex', 'Religious belief', 'Ethnicity',
       'Total UK domiciled students', 'Total', 'Age Group',
       'Disability Status'], dtype=object)

Sex is the only category marker we require for our analysis so we will want to drop the other values.

In [22]:
# drop rows where category marker values not equal to gender

**Category**

We are focusing on gender and need to review the values in this column.

In [23]:
student_df["Category"].unique()

array(['Female', 'Spiritual', 'Sikh', 'Not known', 'Buddhist',
       'Christian', 'Any other religion or belief', 'Hindu', 'Jewish',
       'No religion', 'Muslim', 'Total UK domiciled students', 'Other',
       'Total', '21-24 years', '20 and under', '25-29 years',
       '30 years and over', 'Male', 'Black', 'White',
       'No known disability', 'Asian', 'Mixed', 'Age unknown',
       'Known disability'], dtype=object)

The only categories relating to gender are female, male and other. As our project is focusing on gender, we only want to retain rows with these values for our analysis.

In [34]:
# look to remove rows where the value is not a gender value

**First year marker**

In [25]:
student_df["First year marker"].unique()

array(['All', 'Other years', 'First year'], dtype=object)

First year marker is whether a student is in their first year of study for their degree or other years. For higher education data in our project, we are looking at first year students and then graduates to get an idea of completion, i.e. who starts vs who finishes. 

In the First year marker column, we only want to retain the value of 'First year'. We will want to drop rows with the 'All' and 'Other years' values.

In [26]:
# drop rows where first year marker values not equal to first year.

**Level of study**

In [27]:
student_df["Level of study"].unique()

array(['All', 'Postgraduate (research)', 'Postgraduate (taught)',
       'All undergraduate', 'First degree', 'Other undergraduate',
       'All postgraduate'], dtype=object)

Looking at the values in this column, it would be interesting to see the number and proportion of females starting different levels of degree. 

To get the breakdown, we will remove rows with a value of 'All'. We will need to begin analysing the data to review the merits of retaining the 'All undergraduate' and 'All postgraduate' values. 

Postgraduate combines all postgraduate taught and postgraduate research students, whereas I think we would be better splitting these out. 

'All undergraduate' contains first degrees and other degrees - we may not want to include other degrees in our analysis as it is a broad category where we cannot specify the degree achieved as an outcome.

**Mode of study**

In [28]:
student_df["Mode of study"].unique()

array(['All', 'Full-time', 'Part-time'], dtype=object)

This column contains information on whether students are studying full or part-time. Here I think it will be useful to retain all values initially. We might want to use all to get the general trends but can then look at full-time and part-time split by gender to see if women are more or less likely to study full-time than males. 

**Country of HE provider**

In [29]:
student_df["Country of HE provider"].unique()

array(['All', 'Wales', 'Scotland', 'Northern Ireland', 'England'],
      dtype=object)

Student data is available for the four devolved nations. As the secondary level education data we have is only available for England, we should only retain rows with a value of England for our project.

**Domicile**

In [30]:
student_df["Domicile"].unique()

array(['All', 'Wales', 'Total UK', 'Total Non-UK', 'Scotland', 'Other UK',
       'England', 'Northern Ireland', 'European Union',
       'Non-European Union', 'Not known'], dtype=object)

Our project is focusing on gender in relation to education. We are not looking at where the students are coming from (Although this would be an intersting question to follow on from our project). 

The domicile column can be removed from the dataset.

In [31]:
# remove/ drop domicile column.

**Academic year**

In [33]:
student_df["Academic Year"].unique()

array(['2016/17', '2017/18', '2018/19', '2019/20', '2020/21'],
      dtype=object)

Data is available in this column for the years 2016/7 to 2020/1, which means we will be able to undertake a time series analysis on how many females are entering higher education for the past 5 years. All values will be included